tfjs-backend-webgpu/src/shader_preprocessor.ts

```diff
@@ -222,33 +251,33 @@ function getSetOutputSnippet(
     }`;
   }
 
+  if (!needsShapesUniforms) {
+    return snippet;
+  }
+
   if (outRank >= 2) {
     const dims = ['d0', 'd1', 'd2', 'd3'].slice(0, outRank);
     const type = getCoordsDataType(outRank);
 
     if (isVec4) {
       snippet += `
       void setOutput(${dims.map(d => `int ${d}`).join(', ')}, vec4 value) {
-        int flatIndex = getFlatIndex(${type}(${dims.join(', ')}), ${
-          getShapeCoords(outShape)});
```

Comment:
Don't forget to remove the definition of `getShapeCoords` since it's not needed anymore.

Reply:
Done